pgx-pg-sys/build.rs

```diff
@@ -392,23 +416,632 @@ fn impl_pg_node(
+            if !self.joinrightcols.is_null() {
+                walker_fn(self.joinrightcols as *mut Node, context);
+                Node::traverse::<T>(unsafe { &mut *(self.joinrightcols as *mut Node) }, walker_fn, context);
+            }
+        }
+    } else {
+        quote! {
+            if !self.joinleftcols.is_null() {
+                walker_fn(self.joinleftcols as *mut Node, context);
+                Node::traverse::<T>(unsafe { &mut *(self.joinleftcols as *mut Node) }, walker_fn, context);
+            }
+            if !self.joinrightcols.is_null() {
+                walker_fn(self.joinrightcols as *mut Node, context);
+                Node::traverse::<T>(unsafe { &mut *(self.joinrightcols as *mut Node) }, walker_fn, context);
+            }
+            if !self.join_using_alias.is_null() {
+                walker_fn(self.join_using_alias as *mut Node, context);
+                Node::traverse::<T>(unsafe { &mut *(self.join_using_alias as *mut Node) }, walker_fn, context);
+            }
+        }
+    };
+    let rtekind_result_handling = if std::env::var("CARGO_FEATURE_PG11").is_ok() {
+        quote! {}
+    } else {
+        quote! {
+            RTEKind_RTE_RESULT => {
+                /* no extra fields */
+            },
+        }
     };
 
-    // now we can finally iterate the Nodes and emit out Display impl
-    for node_struct in nodes {
-        let struct_name = &node_struct.struct_.ident;
+    // Older Postgres' List is linked.
+    let list_fns = if std::env::var("CARGO_FEATURE_PG11").is_ok()
+        || std::env::var("CARGO_FEATURE_PG12").is_ok()
+    {
+        quote! {
+            impl pg_sys::PgNode for List {
+                fn traverse<T>(&mut self, walker_fn: fn(*mut Node, &mut T) -> (), context: &mut T) {
+                    if self.length == 0 { return; }
+                    let mut cell = unsafe { *self.head };
+                    for _index in 0..self.length {
+                        let item = unsafe { cell.data.ptr_value as *mut Node };
+                        if !item.is_null() {
+                            walker_fn(item, context);
+                            Node::traverse::<T>( unsafe { &mut *(item as *mut Node) }, walker_fn, context);
+                        }
+                        if !cell.next.is_null() {
+                            cell = unsafe { *cell.next };
+                        }
+                    }
+                }
+            }
+            impl std::fmt::Display for List {
+                fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                    write!(f, "[")?;
+                    if self.length != 0 {
+                        let mut cell = unsafe { *self.head };
+                        for index in 0..self.length {
+                            if index != 0 {
+                                write!(f, ", ")?;
+                            }
+                            let item = unsafe { cell.data.ptr_value as *mut Node };
```

Comment:
What's the precondition that is allowing us to access this as `.ptr_value`?

Reply:
This (and the other instance) were guarded by the "ignore `IntList` and `OidList` clause elsewhere in the code, but that check should be inside this function instead.

Reply:
This is now fixed, specifically for the `List` traversal function.